### PR TITLE
[CPU] Fixed Convert [FP->BOOL]

### PR DESCRIPTION
**Pull request by a-sidorova**
### Details:
 - *Fixed CPU transformation that decomposes `Convert[..->bool]` to `Abs, Ceiling, Convert[..->u8]` - added `Min(Abs, UINT8_MAX)` to avoid overflows*
 - *If `Convert[..->bool]` is inserted on output, the Plugin applies the transformation which is described above and inserts `Convert[u8->bool]` on output to align model output precisions. The `Convert[u8->bool]` is executed by plugin, reference `cpu_convert`. The PR added `bool` lowest and max bounds in `cpu_convert` to have normalized output data - [0, 1] (not [0..256])*
 - *Renamed `sequnce_generator.hpp` -> `sequence_generator.hpp` as @itikhono asked 😃*
 - *Added specific input data generation for `Convert[..->boolean]`*
 - *Added `fill_data_boolean` to generate only 0, 1 values for boolean input data*

### Tickets:
 - *136130, 122321*

### CODE REVIEW

**Comment by a-sidorova** [comment]
@itikhono could you please take a look at transformation related code?

@dmitry-gorokhov could you please review CPU part as well?

**Review by dmitry-gorokhov — APPROVED** [review]
LGTM

**Review by sshlyapn — APPROVED** [review]

File: src/common/transformations/tests/type_relaxed_tests.cpp [hunk 1]
```diff
@@ -411,17 +412,38 @@ bool fuse_type_to_convert_cpu(const std::shared_ptr<ov::Node>& node, const preci
...
+            const auto& in_prec = convert->get_input_element_type(0);
+            auto data = convert->input_value(0).get_node_shared_ptr();
+            auto item = precisions.find(in_prec);
+            if (item != precisions.end()) {
+                // Add convert node for unsupported precision, such as FP64
+                CREATE_OP(pre_convert, ov::opset1::Convert, data, item->second);
+                data = pre_convert;
+            }
+
+            CREATE_OP(abs, ov::opset1::Abs, data);
```
Comment (itikhono) [inline src/common/transformations/tests/type_relaxed_tests.cpp:436]: I have some doubts about using the "CREATE_OP" macro.
I think it makes the code less readable, hard to understand connections between nodes.

I would recommend to use ov::pass::NodeRegistry class, e.g.
``` 
NodeRegistry reg;
auto ov_const = reg.make<ov::opset1::Constant>(...args); // example of shared_ptr<Constant> creation
...
ov::copy_runtime_info(convert, reg.get());
```

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp [hunk 2]
```diff
@@ -216,35 +217,44 @@ bool Transformations::fuse_type_to_convert(const std::shared_ptr<ov::Node>& node
...
-            auto ceil = std::make_shared<ov::opset10::Ceiling>(abs);
-            auto new_convert = std::make_shared<ov::opset10::Convert>(ceil, to);
-            new_convert->set_friendly_name(convert->get_friendly_name());
-            ov::copy_runtime_info(convert, {abs, ceil, new_convert});
-            ov::replace_node(convert, new_convert);
+            CREATE_OP(pre_convert, ov::opset10::Convert, data, item->second);
+            data = pre_convert;
         }
+
+        CREATE_OP(abs, ov::opset10::Abs, data);
```
Comment (itikhono) [inline src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:241]: I have some doubts about using the "CREATE_OP" macro.
I think it makes the code less readable, hard to understand connections between nodes.

I would recommend to use ov::pass::NodeRegistry class, e.g.

```
NodeRegistry reg;
auto ov_const = reg.make<ov::opset1::Constant>(...args); // example of shared_ptr<Constant> creation
ov::copy_runtime_info(convert, reg.get());
```

File: src/common/transformations/tests/type_relaxed_tests.cpp (hunk 1 above)
Comment (a-sidorova) [reply src/common/transformations/tests/type_relaxed_tests.cpp:436]: Cool suggestion! I didn't know about `NodeRegistry`!
Applied, thanks!

File: src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp (hunk 2 above)
Comment (a-sidorova) [reply src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp:241]: Thanks a lot! Done!

**Review by itikhono — APPROVED** [review]

**Pull request merged**